### DISCUSSION TITLE
Amp and non amp pages in dynamic routes

### DISCUSSION THREAD

**Discussion by AngelAngelides** [vercel/next.js#32169] [Ideas]
### Describe the feature you'd like to request

Our application is a magazine application, and we have different types of pages like articles, tags, channels, horoscope pages etc. 

Our business requirements are that some of those pages need to have an am URL using the ?amp=1 parameter, and that some of these pages have no amp available. 

The problem is that they are all handled as dynamic pages, as there is no way to distinguish between two different pages via URL.

Currently, we have no way to activate the amp config on some of these pages and deactivate it on others

### Describe the solution you'd like

We would like to have a way of selectively activating hybrid amp on dynamic pages

### Describe alternatives you've considered

We have been overwriting amphtml links on pages that shouldn't have amp available, but this currently generates errors. We are considering removing amphtml links via JavaScript. If there is another way, we can stop next.js from automatically appending amphtml links please let us know.

**Comment by tomaszrondio** [comment]
same here. It would be great to have an option to set hybrid depending on content behind the dynamic route.

**Comment by buraste** [comment]
Same issue. We need to conditionally select amp page on dynamic routes. Any news?

**Comment by bhumigoklani** [comment]
Same issue.
We are having a news based website where the news detail page needs to have AMP enabled.
The detail page route we have is domain.com/news-slug-here 

We have to load the content of news based on its slug. But when URL is loaded with ?amp=1, we are unable to read the slug so there is no way to load the dynamic article content.

Please suggest.

**Comment by PegeDev** [comment]
any update for this problem?
